packages/web-api-scan-job-manager/src/worker/worker.ts

```diff
@@ -73,12 +74,24 @@ export class Worker {
                 break;
             }
 
-            if (moment().toDate() > this.restartAfterTime) {
+            if (moment().toDate() >= this.restartAfterTime) {
                 this.logger.logInfo(`Performing scheduled termination after ${this.jobManagerConfig.maxWallClockTimeInHours} hours.`);
+                await this.waitForChildTasks();
                 break;
             }
 
-            await System.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+            await this.system.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+        }
+    }
+
+    private async waitForChildTasks(): Promise<void> {
+        this.logger.logInfo('Waiting for child tasks to complete');
+
+        let poolMetricsInfo: PoolMetricsInfo = await this.batch.getPoolMetricsInfo();
+
+        while (poolMetricsInfo.load.activeTasks !== 0 || poolMetricsInfo.load.runningTasks > 1) {
```

Comment:
you can generalize the exit case and use it from above: 

The Batch service API may set activeTasks value instead of runningTasks value hence handle this case
--
  | if (poolMetricsInfo.load.runningTasks + poolMetricsInfo.load.activeTasks === 1) {
  | this.logger.logInfo(`Exiting the ${this.jobId} job since there are no active/running tasks.`);
  | break;
  | }